.clang-tidy

```diff
@@ -48,7 +51,7 @@ Checks: '
   -readability-function-size
   '
 
-HeaderFilterRegex: '.*(/include/(hilti|spicy)/(^3).*/[^_][^_][^/]*$)|(^3rdparty.*)'
```

Comment:
Oh,  that old expression was clearly broken, that `(^3)` doesn't work, that's supposed to be `[^3]`.  However, that was a recent change introduced in 598da755835f878387257f0668e563f1b2255515. Did that change break the headers, or weren't they not working before either?

I'm fine with the new expression too, just wondering if we wouldn't need to be quite as strict ("no digits").